src/Build/Logging/FancyLogger/FancyLoggerNode.cs

```diff
@@ -4,148 +4,136 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
 using Microsoft.Build.Framework;
 
 namespace Microsoft.Build.Logging.FancyLogger
 { 
-    // TODO: Maybe remove
-    public enum FancyLoggerNodeType
+    public class FancyLoggerProjectNode
     {
-        None,
-        Build,
-        Project,
-        Target,
-        Task,
-        Message,
-        Warning,
-        Error
-    }
+        private static string GetUnambiguousPath(string path)
```

Comment:
nit:
Is this function actually needed? I'm wondering why it's better than calling Path.GetFileName() on the relevant paths.

Reply:
Yes. @rainersigwald and I agreed to create a function that for the moment does that but that eventually is able to disambiguate paths (ie, if two projects have the same file name but in different directories)